posixlib/src/main/scala/scala/scalanative/posix/unistd.scala

```diff
@@ -140,18 +157,25 @@ object unistd {
   def nice(inc: CInt): CInt = extern
 
   def pathconf(path: CString, name: CInt): CLong = extern
+
   @blocking
   def pause(): CInt = extern
+
   def pipe(fildes: Ptr[CInt]): CInt = extern
+  def pipe2(fildes: Ptr[CInt], flags: CInt): CInt = extern
```

Comment:
In these 2 pipe functions, these are declared in C like `(int [2])`. So I guess that is an `int` `array` of length 2? `Ptr[CInt]` seems reasonable but just wondering if that should be a `CArray[CInt, _2]` probably with a `Ptr` to wrap it?

Reply:
Thank you for the review and discussion.

The `extern` file is the place where the Scala Native and C cultures meet and bump
shoulders. Sort of like the motorized walkways in airports with strips of differing speed.
Stepping from one to another takes care and can unbalance.

When looked at from the Scala Native side, you have a definite point: get the compiler to
do bounds checking of known sizes. When looked
at from the C side, the current code is quite normal & expected; ptr to Int and address of array
of Int where the Array size is conveyed in another channel, here documentation but often an size_t
argument, is a mental translation made without much effort.

Here, I think extending the existing long standing definition from `pipe(fd)` is the path of least
astonishment. I think `pipe(fd)` and `pipe2(fd, flags)` should be similar. 

I really do not want to change `pipe(fd)`.   If for no other reason in that I think it is used in several places
in SN and I do not want to break other existing users. I have a three  "waiting for 0.6.n" efforts going
already.  

I do have to admit to an underlying dislike of Scala Native `CArray` .  I understand (some of)
its intent & purpose but I find it hard to get the declaration right. Needing to use `foo.at(0)` instead
of `foo()` for array references is just annoying. Gripe, grumble, moan.

Reply:
In several places in the past few (12?) months, most recently using `man7` to lookup the argument
names for `getentropy()`  I have run across a C idiom for arrays of a known size I had not seen before. 
It goes like then like `int foo(void buf[.length], size_t length)`   I think it is doc only and that
one still uses a call like `foo(buf, 2)`.

It must be in a post-C11 standard.  I need to track it down before my next job interview.

Reply:
I agree, that is why I pointed out the Scaladoc is that we could add that the C signature is `int[2]` and that instead of modeling it as `CArray[2]` we used `Ptr[CInt]`. Somewhere along the way that could change but it would have to be a `0.6.0` or something.

Reply:
Good discussion, I appreciate it.  Sorry to be terse now. 

Based on this discussion, and the minimalist approach of Scala, my current thinking
is to add a three or four  line "/** */" Scala Doc mentioning the first argument declaration of
those two is C like, not Scala Native like, referring to this PR for a fuller discussion.
Perhaps giving the Open Group 2024 Issue 8 C snippet.

I concur that it is good to leave a clue to explain what cause some astonishment.
I also think that most readers of that file will be "quick readers" and put off by
page long descriptions of arcane details. Future developers & maintainers 
may appreciate those details & design decisions, so it is is worth capturing them
_someplace_. Since this was supposed to be a quick, low stress, PR so that
I could start using `getentropy()` there is no associated Issue.

Reply:
Eric,

I added the suggested comment, as Scaladoc.  

I think I went beyond 3 or 4 lines but kept it less than a page.
I am not used to writing Scaladoc, so please feel free to offer corrections
and or suggestions.  Thank you.

PS. I have been using `getentropy()` pretty heavily. It and the declaration
      in this file both work. Having that method saves me _a_lot_ of 
      easy to get wrong complexity.